test/async-test-args.js

```diff
@@ -0,0 +1,167 @@
+
+describe('Async parse correctly handling options', () => {
+
+  it('will export source maps', function() {
+    protagonist.parse(data, { exportSourcemap : true })
+      .then(res => {
+        assert.deepEqual(res, sync_parsed_sm);
+      })
+      .catch(err => {
+        assert.fail()
+      })
+  })
+
+  it('will not export source maps', function() {
+    protagonist.parse(data, { exportSourcemap : false })
+      .then(res => {
+        assert.deepEqual(res, sync_parsed);
+      })
+      .catch(err => {
+        assert.fail()
+      })
+  })
+
+  it('will not accept empty options', function() {
+    protagonist.parse(data, {})
+      .then(res => {
+        assert.deepEqual(res, sync_parsed);
+      })
+      .catch(err => {
+        assert.fail()
+      })
+  })
+
+})
+
+
+describe('Async parsing - accepted parameters', () => {
+
+  const source = '# API\n ## GET /\n+ Response 200 (text/plain)\n        Hello world\n'
+  var opt = {}
+
+  describe('Promise', () => {
+    it('will accept source only', (done) => {
+        protagonist.parse(data)
+        .then(res => {
+           assert.isNotNull(res)
+           done()
+        })
+        .catch(err => {
+          done(err)
+        })
+    })
+
+    it('will accept source with options', (done) => {
+      protagonist.parse(source, {exportSourcemap : true})
+        .then(res => {
+            assert.isNotNull(res)
+            done()
+        })
+        .catch(err => {
+            done(err)
+        })
+    })
+
+    it('will not accept swapped params', (done) => {
+      try {
+        protagonist.parse({exportSourcemap : true}, source)
+        .then(res => { assert.fail() })
+      } catch (err) {
+        assert.equal(err, 'TypeError: wrong 1st argument - string expected')
+        done()
+      }
+    })
+
+    it('will not accept no params', (done) => {
+      try {
+        protagonist.parse()
+        .then(res => { assert.fail() })
+      } catch (err) {
+        assert.match(err, /TypeError: wrong number of arguments/)
+        done()
+      }
+    })
+
+    it('will not accept too much params', (done) => {
+      try {
+        protagonist.parse('a','b','c', 'd')
+        .then(res => { assert.fail() })
+      } catch (err) {
+        assert.match(err, /TypeError: wrong number of arguments/)
+        done()
+      }
+    })
+
+    it('will not accept wrong wrong type for options', (done) => {
+      try {
+        protagonist.parse('a','b')
+        .then(res => { assert.fail() })
+      } catch (err) {
+        assert.match(err, /TypeError: wrong 2nd argument - `options` expected/)
+        done()
+      }
+    })
+
+  })
+
+
+  describe('Callback', () => {
+    it('will accept source and callback', (done) => {
+      protagonist.parse(source, (err, res) => {
+        assert.isNull(err)
+        assert.isDefined(res)
+        done()
+      })
+    })
+
+    it('will accept source, options and callback', (done) => {
```

Comment:
This is already tested in `options-test.coffee`

Reply:
Same as above